src/reconcile/rec_write.c

```diff
@@ -2830,14 +2831,22 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
     case 0: /* Page delete */
         WT_STAT_CONN_DSRC_INCR(session, rec_page_delete);
 
+        /*
+         * TODO: We need to tell the PALI interface this page is discarded. Mark it as invalid for
+         * now.
+         */
+        ref->page->block_meta.page_id = WT_BLOCK_INVALID_PAGE_ID;
+
         /*
          * If this is the root page, we need to create a sync point. For a page to be empty, it has
          * to contain nothing at all, which means it has no records of any kind and is durable.
          */
         ref = r->ref;
         if (__wt_ref_is_root(ref)) {
             __wt_checkpoint_tree_reconcile_update(session, &ta);
-            WT_RET(bm->checkpoint(bm, session, NULL, NULL, btree->ckpt, false));
+            WT_RET(bm->checkpoint(
+              bm, session, NULL, &r->wrapup_checkpoint_block_meta, btree->ckpt, false));
+            r->ref->page->block_meta = r->wrapup_checkpoint_block_meta;
```

Comment:
How does the original `page->block_meta` get freed?

Reply:
It is part of the page and it is freed when page is evicted.